baremetal_network_provisioning/drivers/snmp_client.py

```diff
@@ -0,0 +1,143 @@
+                self.priv_key = priv_key
+        else:
+            self.write_community = write_community
+        self.cmd_gen = cmdgen.CommandGenerator()
+
+    def _get_auth(self):
+        """Return the authorization data for an SNMP request.
+
+        """
+        if self.access_protocol == constants.SNMP_V3:
+            return cmdgen.UsmUserData(self.security_name,
+                                      authKey=self.auth_key,
+                                      privKey=self.priv_key,
+                                      authProtocol=self.auth_protocol,
+                                      privProtocol=self.priv_protocol)
+        else:
+            mp_model = 1 if self.access_protocol == constants.SNMP_V2C else 0
+            return cmdgen.CommunityData(self.write_community,
+                                        mpModel=mp_model)
+
+    def _get_transport(self):
+        """Return the transport target for an SNMP request.
+
+        """
+        return cmdgen.UdpTransportTarget(
+            (self.ip_address, constants.SNMP_PORT), timeout=1, retries=5)
+
+    def get(self, oid):
+        """Use PySNMP to perform an SNMP GET operation on a single object.
+
+        """
+        try:
+            results = self.cmd_gen.getCmd(self._get_auth(),
+                                          self._get_transport(),
+                                          oid)
+        except snmp_error.PySnmpError as e:
+            raise exception.SNMPFailure(operation="GET", error=e)
+
+        error_indication, error_status, error_index, var_binds = results
+
+        if error_indication:
+            raise exception.SNMPFailure(operation="GET",
+                                        error=error_indication)
+
+        if error_status:
+            raise exception.SNMPFailure(operation="GET",
+                                        error=error_status.prettyPrint())
+
+        return var_binds
+
+    def get_bulk(self, *oids):
+        try:
+            results = self.cmd_gen.bulkCmd(self._get_auth(),
+                                           self._get_transport(),
+                                           0, 52,
+                                           *oids
+                                           )
+        except snmp_error.PySnmpError as e:
+            raise exception.SNMPFailure(operation="GET", error=e)
+
+        error_indication, error_status, error_index, var_binds = results
+
+        if error_indication:
+            raise exception.SNMPFailure(operation="GET",
+                                        error=error_indication)
+
+        if error_status:
+            raise exception.SNMPFailure(operation="GET",
+                                        error=error_status.prettyPrint())
+
+        return var_binds
+
+
+def get_client(snmp_info):
+    """Create and return an SNMP client object.
+
+    """
+    return SNMPClient(snmp_info["ip_address"],
+                      snmp_info["access_protocol"],
```

Comment:
why it is different for accessing the (ip_address,access_protocol,write_community ) & (security_name ,etc) ?